iroh-bytes/src/util.rs

```diff
@@ -252,4 +254,15 @@ mod tests {
         let encoded = hash.to_string();
         assert_eq!(encoded.parse::<Hash>().unwrap(), hash);
     }
+
+    #[test]
+    fn hash_wire_format() {
+        let hash = Hash::from([0xab; 32]);
+        let serialized = postcard::to_stdvec(&hash).unwrap();
+        let expected = parse_hexdump(r"
+            20 # length prefix
```

Comment:
This was unexpected. I thought hashes would be serialized as 32 bytes without length prefix. Should we get rid of the length prefix?